src/coreComponents/physicsSolvers/fluidFlow/DissipationCompositionalMultiphaseFVMKernels.hpp

```diff
@@ -0,0 +1,395 @@
+                                           localIndex const er_up,
+                                           localIndex const esr_up,
+                                           localIndex const ei_up,
+                                           real64 const & potGrad,
+                                           real64 const & phaseFlux,
+                                           real64 const (&dPhaseFlux_dP)[2],
+                                           real64 const (&dPhaseFlux_dC)[2][numComp] )
+    {
+      GEOS_UNUSED_VAR( k_up, potGrad, phaseFlux, dPhaseFlux_dP, dPhaseFlux_dC, er_up, esr_up, ei_up );
+
+      /// Storing dissipation flux and its derivatives locally
+      real64 dissFlux[numComp]{};
+      real64 dDissFlux_dP[numFluxSupportPoints][numComp]{};
+      real64 dDissFlux_dC[numFluxSupportPoints][numComp][numComp]{};
+      real64 fluxPointCoef[numFluxSupportPoints] = {1.0, -1.0}; // for gradients
+
+      real64 viscosityMult[3] = {1.0, 1.0, 1.0}; // for viscosity
+
+      /// Step 2. Collect all contributions
+      real64 poreVolume_n = 0.0; // Pore volume contribution
+      real64 trans = 0.0; // Transmissibility contribution
+      for( integer ke = 0; ke < numFluxSupportPoints; ++ke )
+      {
+        trans = std::max( stack.transmissibility[connectionIndex][ke], trans );
+        poreVolume_n += m_volume[seri[ke]][sesri[ke]][sei[ke]] * m_porosity_n[seri[ke]][sesri[ke]][sei[ke]][0];
+      }
+      poreVolume_n /= numFluxSupportPoints;
+
+      // potential gradient contribution
+      // pressure
+      real64 pressure_gradient = 0;
+      for( integer ke = 0; ke < numFluxSupportPoints; ++ke )
+        pressure_gradient += fluxPointCoef[ke] * m_pres_n[seri[ke]][sesri[ke]][sei[ke]];
+
+      real64 const potential_gradient = LvArray::math::abs( pressure_gradient );
+
+      real64 grad_depth = 0;
+      for( integer ke = 0; ke < numFluxSupportPoints; ++ke )
+      {
+        grad_depth += fluxPointCoef[ke] * m_gravCoef[seri[ke]][sesri[ke]][sei[ke]];
+      }
+
+      // bias towards x and y direction for miscible
+      /* The point of this "directional coefficient" was to add less dissipation flux in the direction of gravity and 
+      more in all other directions.  These hard-coded values of 1000 and 100 were picked by some tuning and as 
+      expected are not robust. Further research/testing is needed to generalize this idea.
+      */
+      real64 directional_coef = 1.0;
+      if( m_miscibleDBC )
+      {
+        directional_coef = 100.0;
+        if( LvArray::math::abs( grad_depth ) > 0.0 )
+        {
+          real64 const d2 = LvArray::math::abs( grad_depth * grad_depth );
+          if( 1000.0 / d2 < 100.0 )
+            directional_coef = 1000.0 / d2;
+        }
+      }
+
+      // multiplier with all contributions
+      real64 const multiplier = m_kappaDBC * m_omegaDBC * trans / poreVolume_n * m_dt * potential_gradient * directional_coef;
```

Comment:
Following above comment `trans` should be abs closer to this point

Reply:
> Following above comment `trans` should be abs closer to this point

fixed